toml/src/main/kotlin/org/rust/toml/completion/LocalCargoTomlDependencyCompletionProvider.kt

```diff
@@ -37,53 +40,130 @@ class LocalCargoTomlDependencyCompletionProvider : TomlKeyValueCompletionProvide
                 LookupElementBuilder
                     .create(crateName)
                     .withIcon(AllIcons.Nodes.PpLib)
-                    .withInsertHandler { ctx, _ ->
-                        val alreadyHasValue = ctx.nextCharIs('=')
-
-                        if (!alreadyHasValue) {
-                            ctx.document.insertString(ctx.selectionEndOffset, " = \"\"")
-                        }
-
-                        EditorModificationUtil.moveCaretRelatively(ctx.editor, 4)
-
-                        if (!alreadyHasValue) {
-                            // Triggers dependency version completion
-                            AutoPopupController.getInstance(ctx.project).scheduleAutoPopup(ctx.editor)
-                        }
-                    },
+                    .withInsertHandler(KeyInsertHandlerWithCompletion()),
                 (-crateName.length).toDouble()
             )
         }
         result.withPrefixMatcher(CargoDependenciesPrefixMatcher(prefix)).addAllElements(elements)
     }
 
     override fun completeValue(keyValue: TomlKeyValue, result: CompletionResultSet) {
-        val name = CompletionUtil.getOriginalElement(keyValue.key)?.text ?: return
+        val keySegment = keyValue.key.segments.singleOrNull() ?: return
+        val name = CompletionUtil.getOriginalElement(keySegment)?.text ?: return
```

Comment:
Could you please add a test for these change? Currently, when I change these lines back, all tests pass